AllEvents.TicketManagement/src/Core/AllEvents.TicketManagement.Application/Features/Tickets/Handlers/GenerateTicketCommandHandler.cs

```diff
@@ -0,0 +1,120 @@
+using AllEvents.TicketManagement.Domain.Entities;
+using MediatR;
+using QRCoder;
+using System.Drawing.Imaging;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace AllEvents.TicketManagement.Application.Features.Tickets.Handlers
+{
+    public class GenerateTicketCommandHandler : IRequestHandler<GenerateTicketCommand, TicketModel>
+    {
+        private readonly IEventRepository _eventRepository;
+        private readonly ITicketRepository _ticketRepository;
+        private readonly byte[] aesKey = Encoding.UTF8.GetBytes("A3C9F7E20B567891"); 
+        private readonly byte[] aesIV = Encoding.UTF8.GetBytes("E1F5D1A2C9B81234");
+
+        public GenerateTicketCommandHandler(IEventRepository eventRepository, ITicketRepository ticketRepository)
+        {
+            _eventRepository = eventRepository ?? throw new ArgumentNullException(nameof(eventRepository));
+            _ticketRepository = ticketRepository ?? throw new ArgumentNullException(nameof(ticketRepository));
+        }
+
+        public async Task<TicketModel> Handle(GenerateTicketCommand request, CancellationToken cancellationToken)
+        {
+            var @event = await _eventRepository.GetByIdAsync(request.EventId);
+
+            if (@event == null)
+            {
+                throw new ArgumentException($"Event with ID {request.EventId} not found.");
+            }
+
+            var ticketId = Guid.NewGuid();
+            var encryptedData = EncryptData($"{ticketId}:{request.PersonName}");
+            var qrCodeImage = GenerateQRCodeImage(encryptedData);
+
+            var ticket = new Ticket(
+                ticketId: ticketId,
+                personName: request.PersonName,
+                eventTitle: @event.Title,
+                qRCode: qrCodeImage,
+                eventId: @event.EventId
+            );
+
+            await _ticketRepository.AddAsync(ticket);
+
+            return new TicketModel(
+                ticketId: ticket.TicketId,
+                personName: ticket.PersonName,
+                eventTitle: ticket.EventTitle,
+                qRCode: ticket.QRCode
+            );
+        }
+
+        private byte[] EncryptData(string plainText)
+        {
+            using (Aes aes = Aes.Create())
+            {
+                aes.Key = aesKey;
+                aes.IV = aesIV;
+                ICryptoTransform encryptor = aes.CreateEncryptor(aes.Key, aes.IV);
+
+                using (var ms = new MemoryStream())
+                {
+                    using (var cs = new CryptoStream(ms, encryptor, CryptoStreamMode.Write))
+                    {
+                        using (var sw = new StreamWriter(cs))
+                        {
+                            sw.Write(plainText);
+                        }
+                        return ms.ToArray();
+                    }
+                }
+            }
+        }
+
+        private string DecryptData(byte[] cipherText)
```

Comment:
We are not going to need this one in this Handler so please delete it 
@KKDinev20